File: LeetCode_Csharp/LeetCode_Csharp/Code/StackAndQueue/LeetCode150_EvalRPN_middle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace LeetCode_Csharp.Code.StackAndQueue
{
    public class LeetCode150_EvalRPN_middle
    {
        public int EvalRPN(string[] tokens)
        {
            Stack<int> stack = new Stack<int>();
            // HashSet<char> charSet = new HashSet<char>{'+', '-', '*', '/'};
            for(int i = 0; i < tokens.Length; i++)
            {
                if(int.TryParse(tokens[i], out int num))
                {
                    stack.Push(num);
                }
                else
                {
                    int num1 = stack.Pop();
                    int num2 = stack.Pop();
                    int result = tokens[i] switch
                    {
                        "+" => num1 + num2,
                        "-" => num2 - num1,
                        "*" => num1 * num2,
                        "/" => num2 / num1,
                        _ => 0
                    };
                    stack.Push(result);
                }
            }
            return stack.Pop();
        }
    }
}

File: LeetCode/LeetCode/LeetCode150_EvalRPN_middle.cpp
#include <stack>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stack;
		for (string token : tokens)
		{
			if (token == "+" || token == "-" || token == "*" || token == "/")
			{
				int num2 = stack.top();
				stack.pop();
				int num1 = stack.top();
				stack.pop();
				if (token == "+")
				{
					stack.push(num1 + num2);
				}
				else if (token == "-")
				{
					stack.push(num1 - num2);
				}
				else if (token == "*")
				{
					stack.push(num1 * num2);
				}
				else if (token == "/")
				{
					stack.push(num1 / num2);
				}
			}
			else
			{
				stack.push(stoi(token));
			}

		}

		int result = stack.top();
		stack.pop();
		return result;
    }
};